lib/business_time/config.rb

```diff
@@ -127,6 +134,25 @@ def weekdays
         end
       end
 
+      def load_holidays(holidays, container: config[:holidays], append: false, accept_parsing: false)
+        container.replace(DEFAULT_CONFIG[:holidays]) unless append
+        holidays.each do |holiday|
+          container.add(if Date === holiday
+            holiday
+          else
+            Date.parse(holiday).tap do |parsed|
+              ActiveSupport::Deprecation.warn("Provide holidays as `Date` objects instead of `#{holiday.class}`. I parsed that thing as #{parsed}", caller) unless accept_parsing
+            end
+          end)
```

Comment:
The whole `container.add` is looking like a rocket science while it is not. I would do (not forgetting that `container.is_a?(Set) == true`:
```ruby
container |= holidays.map do |holiday|
  ActiveSupport::Deprecation.warn( .... )
  Date.parse(holiday.to_s) rescue nil
end.compact
```